### PR TITLE
Ctcp 4062

### PR DESCRIPTION
**Pull request by derekspalmer**

### CODE REVIEW

File: app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala [hunk 1]
```diff
@@ -24,8 +24,11 @@ import cats.data.EitherT
 import play.api.Logging
 import play.api.http.MimeTypes
 import play.api.libs.Files.TemporaryFileCreator
+//import play.api.libs.json.Format.GenericFormat
 import play.api.libs.json.JsError
+import play.api.libs.json.JsObject
 import play.api.libs.json.Json
+//import play.api.libs.json.OFormat.oFormatFromReadsAndOWrites
```
Comment (drnaylor) [inline app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala]: If they're not used, let's bin them.

File: app/uk/gov/hmrc/transitmovementsauditing/config/Constants.scala [hunk 2]
```diff
@@ -17,5 +17,12 @@
...
 object Constants {
-  val XContentLengthHeader = "X-ContentLength"
+  val XContentLengthHeader   = "X-ContentLength"
+  val XAuditMetaPath         = "X-Audit-Meta-Path"
+  val XAuditMetaMovementId   = "X-Audit-Meta-Movement-Id"
+  val XAuditMetaMessageId    = "X-Audit-Meta-Message-Id"
+  val XAuditMetaEORI         = "X-Audit-Meta-EORI"
+  val XAuditMetaMovementType = "X-Audit-Meta-Movement-Type"
+  val XAuditMetaMessageType  = "X-Audit-Meta-Message-Type"
+  val XAuditSource           = "X-Audit-Source"
```
Comment (drnaylor) [inline app/uk/gov/hmrc/transitmovementsauditing/config/Constants.scala]: Note that #21 also does this, so you might have a conflict to deal with.

File: app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala [hunk 3]
```diff
@@ -114,6 +139,21 @@ class AuditController @Inject() (
         presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError))
       )
 
+  //TODO: combine parseObj and parseDetails??
+  //      e.g. private def parseXX[A](body: String): EitherT[Future, PresentationError, A]= ???
+  //      requires an implicit reads for the validate!
```
Comment (drnaylor) [inline app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala]: Very much agree we should do this, the implicit shouldn't be an issue:

```scala
private def parse[A: Reads](body: String): EitherT[Future, PresentationError, A] = ???
```

should do the trick.

File: app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala [hunk 4]
```diff
@@ -147,8 +187,9 @@ class AuditController @Inject() (
   ): EitherT[Future, ConversionError, Source[ByteString, _]] =
     if (request.contentType.contains(MimeTypes.XML) && auditType.messageType.isDefined)
       conversionService.toJson(auditType.messageType.get, request.body)
-    else
+    else {
```
Comment (drnaylor) [inline app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala]: We should put braces on the if above too.

File: app/uk/gov/hmrc/transitmovementsauditing/services/AuditService.scala [hunk 5]
```diff
@@ -48,6 +49,11 @@ import scala.util.control.NonFatal
 @ImplementedBy(classOf[AuditServiceImpl])
 trait AuditService {
 
+  def sendMessageTypeEvent(auditType: AuditType, details: Details)(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, AuditError, Unit]
+
+  // TODO: still required??
```
Comment (drnaylor) [inline app/uk/gov/hmrc/transitmovementsauditing/services/AuditService.scala]: Good question, once Prashanth's stuff is in, if it's not needed, bin it.

**Review by drnaylor — COMMENTED** [review]
Few thoughts.

File: app/uk/gov/hmrc/transitmovementsauditing/services/AuditService.scala (hunk 5 above)
Comment (derekspalmer) [reply app/uk/gov/hmrc/transitmovementsauditing/services/AuditService.scala]: Removed.

**Review by drnaylor — APPROVED** [review]
Looks reasonable to me.

**Review by prashanthmamidi — APPROVED** [review]

**Pull request merged**